ui/src/components/credentials/IssueCredential.tsx

```diff
@@ -1,122 +1,142 @@
+      ].join("\n")
+    : `An error occurred while downloading the json schema from the URL:\n"${error}"\nPlease try again.`;
+
 export function IssueCredential() {
   const env = useEnvContext();
-  const [credential, setCredential] = useState<AsyncTask<OldCredential, undefined>>({
+
+  const [step, setStep] = useState<Step>("issuanceMethod");
+  const [credentialFormInput, setCredentialFormInput] = useState<CredentialFormInput>(
+    defaultCredentialFormInput
+  );
+  const [schema, setSchema] = useState<Schema>();
+  const [jsonSchema, setJsonSchema] = useState<AsyncTask<JsonSchema, string | z.ZodError>>({
     status: "pending",
   });
-  const [formData, setFormData] = useState<FormData>(defaultFormData);
-  const [schema, setSchema] = useState<AsyncTask<Schema, APIError>>({
+  const [linkID, setLinkID] = useState<AsyncTask<string, undefined>>({
     status: "pending",
   });
 
-  const [step, setStep] = useState<IssuanceStep>("attributes");
-
   const { schemaID } = useParams();
 
-  const fetchSchema = useCallback(
-    async (signal: AbortSignal) => {
-      if (schemaID) {
-        setSchema({ status: "loading" });
+  const processError = (error: unknown) =>
+    error instanceof z.ZodError ? error : error instanceof Error ? error.message : "Unknown error";
 
-        const response = await getSchema({
-          env,
-          schemaID,
-          signal,
-        });
-
-        if (response.isSuccessful) {
-          setSchema({ data: response.data, status: "successful" });
-        } else {
-          if (!isAbortedError(response.error)) {
-            setSchema({ error: response.error, status: "failed" });
-          }
-        }
-      }
-    },
-    [env, schemaID]
-  );
-
-  const issueCredential = (formData: FormData) => {
+  const createLinkFromCredentialLinkForm = (credentialLinkForm: CredentialLinkForm) => {
     if (schemaID) {
-      const parsedForm = getIssueCredentialFormDataParser([]).safeParse(formData);
-
-      if (parsedForm.success) {
-        setCredential({ status: "loading" });
-
-        void credentialIssue({
+      setLinkID({ status: "loading" });
+      const serializedCredentialForm = serializeCredentialLinkForm({
+        issueCredential: credentialLinkForm,
+        schemaID,
+      });
+
+      if (serializedCredentialForm.success) {
+        void createLink({
           env,
-          payload: serializeCredentialForm(parsedForm.data),
-          schemaID,
+          payload: serializedCredentialForm.data,
         }).then((response) => {
           if (response.isSuccessful) {
-            setCredential({ data: response.data, status: "successful" });
+            setLinkID({ data: response.data.id, status: "successful" });
             setStep("summary");
 
             void message.success("Credential link created");
           } else {
-            setCredential({ error: undefined, status: "failed" });
+            setLinkID({ error: undefined, status: "failed" });
```

Comment:
Why do we need to make undefined keys `undefined` explicitly?

Reply:
The error in `AsyncTaskFailed` is not optional.

Reply:
But here it is bypassing the type checking to essentially become optional. Why are we not passing a string here, at least `response.error.message`? Also the same in [`SelectSchema`](https://github.com/0xPolygonID/sh-id-platform/blob/PID-429-issuer-can-submit-a-credential-link-issuance/ui/src/components/credentials/SelectSchema.tsx#L45).

Reply:
Amend pushed as discussed.